pkg/controller/prpqr_reconciler/prpqr_reconciler.go

```diff
@@ -425,16 +510,104 @@ func generateProwjob(rc injectingResolverClient, defaulter periodicDefaulter, ba
 	}
 	periodic.ExtraRefs = []prowv1.Refs{extraRefs}
 
-	if err = defaulter.DefaultPeriodic(periodic); err != nil {
+	if err := defaulter.DefaultPeriodic(periodic); err != nil {
 		return nil, fmt.Errorf("failed to default the ProwJob: %w", err)
 	}
 
 	pj := pjutil.NewProwJob(pjutil.PeriodicSpec(*periodic), labels, annotations)
 	pj.Namespace = prpqrNamespace
 
-	return &pj, err
+	return &pj, nil
 }
 
 func metadataFromPullRequestUnderTest(pr v1.PullRequestUnderTest) *api.Metadata {
 	return &api.Metadata{Org: pr.Org, Repo: pr.Repo, Branch: pr.BaseRef}
 }
+
+func generateAggregatedProwjobs(uid string, ciopConfig *api.ReleaseBuildConfiguration, defaulter periodicDefaulter, baseCiop *api.Metadata, prpqrName, prpqrNamespace string, spec *v1.ReleaseJobSpec, pr *v1.PullRequestUnderTest) ([]*prowv1.ProwJob, error) {
+	var ret []*prowv1.ProwJob
+
+	opts := options{labels: map[string]string{"release.openshift.io/aggregation-id": uid}}
+
+	for i := 0; i < spec.AggregatedCount; i++ {
+		jobName := fmt.Sprintf("%s-%d", spec.JobName(jobconfig.PeriodicPrefix), i)
+
+		pj, err := generateProwjob(ciopConfig, defaulter, baseCiop, prpqrName, prpqrNamespace, spec, pr, jobName, opts)
+		if err != nil {
+			return nil, fmt.Errorf("failed to create prowjob: %w", err)
+		}
+
+		ret = append(ret, pj)
+	}
+
+	return ret, nil
+}
+
+func createJobRunAggregatorJob(uid, aggregatorJobName, jobName, prpqrName, prpqrNamespace string, defaulter periodicDefaulter, startTime time.Time) (*prowv1.ProwJob, error) {
+	//jobSpec.JobName(jobconfig.PeriodicPrefix)
+	labels := map[string]string{"release.openshift.io/aggregation-id": uid, v1.PullRequestPayloadQualificationRunLabel: prpqrName}
+	spec := &corev1.PodSpec{
+		Volumes: []corev1.Volume{
+			{
+				Name: "pull-secret",
+				VolumeSource: corev1.VolumeSource{
+					Secret: &corev1.SecretVolumeSource{SecretName: "registry-pull-credentials"},
+				},
+			},
+			{
+				Name: "gcs-credentials",
+				VolumeSource: corev1.VolumeSource{
+					Secret: &corev1.SecretVolumeSource{SecretName: "gce-sa-credentials-gcs-publisher"},
+				},
+			},
+		},
+		Containers: []corev1.Container{
+			{
+				Image:           "registry.ci.openshift.org/ci/job-run-aggregator",
+				ImagePullPolicy: corev1.PullAlways,
+				Resources: corev1.ResourceRequirements{
+					Requests: corev1.ResourceList{"cpu": *resource.NewMilliQuantity(10, resource.DecimalSI)},
+				},
+				Command: []string{"job-run-aggregator"},
+				Args: []string{
+					"--analyze-job-runs",
+					"--google-service-account-credential-file=/secrets/gcs/service-account.json",
+					fmt.Sprintf("--job=%s", jobName),
+					fmt.Sprintf("--aggregation-id=%s", uid),
+					// TODO: determine how to get the correct gcs prefix for a job
+					// Now this value should be wrong.
+					fmt.Sprintf("--explicit-gcs-prefix=logs/%s", jobName),
```

Comment:
Yep the `--job=` and `--explicit-gcs-prefix` will be different. `--job` is the original name of the job that we mimic, the value in `--explicit-gcs-prefix` is the name of th job that we submit, on L481

```go
periodic.Name = fmt.Sprintf("%s-%s-%d%s-%s", baseCiop.Org, baseCiop.Repo, pr.PullRequest.Number, variant, inject.Test)
```